CalculatorTests/CalculatorTests/Tests/SettingsPageTests.cs

```diff
@@ -0,0 +1,180 @@
+            rateOfInterestField.SendKeys("100");
+            investmentTermField.SendKeys("365");
+            startDayDropdown.SelectByText("19");
+            startMonthDropdown.SelectByText("May");
+            startYearDropdown.SelectByText("2022");
+            financialYearButton365.Click();
+            new WebDriverWait(driver, TimeSpan.FromSeconds(20))
+            .Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(By.Id("calculateBtn")));
+            calculateButton.Click();
+            new WebDriverWait(driver, TimeSpan.FromSeconds(20))
+            .Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(By.Id("calculateBtn")));
+
+            Assert.AreEqual(amountCurrency, amountCurrencySign.GetAttribute("textContent"));
+            Assert.AreEqual(interestEarned, interestEarnedField.GetAttribute("value"));
+            Assert.AreEqual(income, incomeField.GetAttribute("value"));
+            Assert.AreEqual(endDate, endDateField.GetAttribute("value"));
+        }
+
+        [TestCase("dd/MM/yyyy", "123,456,789.00", "$ - US dollar", "$", "100,000.00", "200,000.00", "19/05/2023")]
+        [TestCase("dd-MM-yyyy", "123.456.789,00", "$ - US dollar", "$", "100,000.00", "200,000.00", "19/05/2023")]
+        [TestCase("MM/dd/yyyy", "123 456 789.00", "$ - US dollar", "$", "100,000.00", "200,000.00", "19/05/2023")]
+        [TestCase("MM dd yyyy", "123 456 789,00", "$ - US dollar", "$", "100,000.00", "200,000.00", "19/05/2023")]
+
+
+        public void CancelButtonTests(string dateFormat, string numberFormat, string defaultCurrency, string amountCurrency, string interestEarned, string income, string endDate)
+        {
+            IWebElement settingsButton = driver.FindElement(By.XPath("//div[@class = 'settings link btn btn-link']"));
+
+
+            settingsButton.Click();
+            new WebDriverWait(driver, TimeSpan.FromSeconds(20))
+            .Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(By.Id("save")));
+
+            IWebElement dateFormatField = driver.FindElement(By.Id("dateFormat"));
+            IWebElement numberFormatField = driver.FindElement(By.Id("numberFormat"));
+            IWebElement defaultCurrencyField = driver.FindElement(By.Id("currency"));
+            IWebElement cancelButton = driver.FindElement(By.Id("cancel"));
+            SelectElement dateFormatDropdown = new SelectElement(dateFormatField);
+            SelectElement numberFormatDropdown = new SelectElement(numberFormatField);
+            SelectElement defaultCurrencyDropdown = new SelectElement(defaultCurrencyField);
+
+            dateFormatDropdown.SelectByText(dateFormat);
+            numberFormatDropdown.SelectByText(numberFormat);
+            defaultCurrencyDropdown.SelectByText(defaultCurrency);
+            cancelButton.Click();
+            
+            IWebElement depositAmountField = driver.FindElement(By.XPath("//tr[contains( string(), 'Deposit Amount')]//input"));
```

Comment:
I suggest to go back to the Settings page and verify the values in dropdowns are equal to default ones.

Reply:
Done